### ISSUE TITLE
possible to show a picture in mini media player when product:TV

### ISSUE THREAD

**Issue by elmattt** [thlucas1/homeassistantcomponent_soundtouchplus#14]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/issues?q=is%3Aissue+label%3A%22Feature+Request%22+).

### Is your feature request related to a problem? Please describe.

i use mini media player to show on screens what is playing, it works with BT, spotify, tunenin etc with full cover

### Describe the solution you'd like

would like to add the possibility of showing picture or screencap of my fire tv when the media player is on PRODUCT:TV

### Describe alternatives you've considered

...

### Additional context

...

**Comment by thlucas1** [comment]
Hi @elmattt - thanks for reaching out.  

I don't believe I can help too much with this, as the mini media player (separate integration) is responsible for displaying of cover art and media controls.  The SoundTouchPlus just interfaces with the speaker.  You might try submitting this to the [Mini Media Player GitHub Issues](https://github.com/kalkih/mini-media-player/issues) link to see if they can add the requested functionality.

The best I could do would be to add a service to retrieve the "NowPlaying" information, and return it as service response data.  The information returned might contain a cover art url, which could be used by the mini-media player for display.  Though I am not sure if this would return a cover art image for the "PRODUCT:TV" source.  Here is what I see for my ST-300 NowPlaying when the source is "PRODUCT:TV" ...

command URL = `http://192.168.1.x:8090/nowPlaying`
```
<nowPlaying deviceID="E8EB11B9B723" source="PRODUCT" sourceAccount="TV">
  <ContentItem source="PRODUCT" sourceAccount="TV" isPresetable="false" />
  <art artImageStatus="SHOW_DEFAULT_IMAGE" />
  <playStatus>PLAY_STATE</playStatus>
</nowPlaying>
```

What do you see on your SoundTouch device when browsing to that URL (change device IP address of course) from a desktop browser?

**Comment by elmattt** [comment]
hello and thanks for the answer !
i see the same on my st300 ;)
i'm looking for another solution then !

**Comment by thlucas1** [comment]
Another idea would be to interface with the Fire TV device to see if you can pull a cover art image from it, or maybe get the TV channel (or even network logo) covert art / image.  I do see that [Fire TV integration](https://www.home-assistant.io/integrations/fire_tv/) is available, supported via [Android Debug Bridge integration](https://www.home-assistant.io/integrations/androidtv), though not familiar with that.  In a quick read of ADB, there is a [Use Screen Capture For Album Art](https://www.home-assistant.io/integrations/androidtv#use-screen-capture-for-album-art) function, but not sure how it works.

Another approach would be to interface with some sort of TV EPG (Electronic Programming Guide) to get a cover art image of the TV show currently playing in your time zone for the channel specified.

Keep in mind that the above ideas are outside the scope of SoundTouchPlus; BUT ....
If you could provide an image url, I could add a service to the SoundTouchPlus integration to pass that image url into the MediaPlayerEntity media_image_url property, which should get forwarded on to mini media player.

Hope it helps!

**Comment by thlucas1** [comment]
I added a new custom service to the integration called `update_source_nowplayingstatus`, which allows you to specify a source (e.g. "Product (TV)") and set a bunch of NowPlaying attributes that will show up in the mini-media player for that specific source.  For most sources the SoundTouch device will communicate the current status back to the medi aplayer (e.g. Spotify, TunieIn, Pandora, BlueTooth, AirPlay, etc).  But not for the "PRODUCT" type of sources (e.g. TV, HDMI_1, etc).  This will allow you to update the NowPlayingStatus for those specific sources if you want.  It may require some automations on your end though, as you have to detect when content on the source changes, and call the service with the updates.

The NowPlayingStatus attributes include: album:str, artist:str, artistId:str, artUrl:str, description:str, duration:int, genre:str, playStatus:str, position:int, sessionId:str, stationLocation:str, stationName:str, track:str, trackId.

Example calling the service.
```
  - service: soundtouchplus.update_source_nowplayingstatus
    data:
      entity_id: media_player.bose_st300
      source_title: 'Product (TV)'
      art_url: "{{ state_attr('media_player.kodi_hearthroom', 'entity_picture') }}"
      station_name: "{{ state_attr('media_player.kodi_hearthroom', 'media_title') }}"
      station_location: Kodi Hearthroom
```

I am now in the process of adding some blueprints to call the update service automatically to update the media player status as different types of media (e.g. Movies, TVShow, Music) are played on my Kodi media center, which uses the "Product (TV)" source).

Will keep you posted when I have something ready for release.

**Comment by thlucas1** [comment]
I just published release v1.0.31 that supports manual updates of the NowPlayingStatus.  This will allow you to manually update the `Product (TV)` source with various information.  Check out the [SoundTouchPlus project Service Details wiki](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/wiki/Services-Provided#update-source-nowplayingstatus) page for details of attributes that can be set as well as examples.

I tested it using my Kodi media player to push updates to the SoundTouchPlus player when Kodi attribute changes were detected.  I also setup a [My Home Assistant Blueprints](https://github.com/thlucas1/homeassistant_blueprints) repository for some blueprints related to using this with Kodi.

Check those out and let me know if it helps.
